### PR TITLE
Fix invalid diagnostic for callers

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1293

The caller will validate the following:
- caller name should start with the service name
- caller name should end with the `Caller` suffix

Return type name cannot be validated because there is no return type when we use callers.

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: grpc-native/src/main/java/org/ballerinalang/net/grpc/GrpcConstants.java [hunk 1]
```diff
@@ -277,4 +277,25 @@ private GrpcConstants() {
...
     public static final String CONTENT_FIELD = "content";
     public static final String HEADER_FIELD = "headers";
+
+    // Compiler plugin constants
+    public static final String GRPC_ANNOTATION_NAME = "ServiceDescriptor";
+    public static final String BALLERINA_ORG_NAME = "ballerina";
+    public static final String GRPC_PACKAGE_NAME = "grpc";
+
+    // Diagnostic error messages and IDs
+    public static final String UNDEFINED_ANNOTATION_MSG = "undefined annotation: ";
```
Comment (aashikam) [inline grpc-native/src/main/java/org/ballerinalang/net/grpc/GrpcConstants.java:287]: Error messages and error codes are coupled, right? We can have an enum? ([graphql](https://github.com/ballerina-platform/module-ballerina-graphql/blob/5549699ae7f15f4abf498d3694fe256b23c31546/graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/PluginConstants.java#L40))

Comment (shafreenAnfar) [reply grpc-native/src/main/java/org/ballerinalang/net/grpc/GrpcConstants.java:287]: Yeah, enums would be better.

Comment (BuddhiWathsala) [reply grpc-native/src/main/java/org/ballerinalang/net/grpc/GrpcConstants.java]: Fixed in herehttps://github.com/ballerina-platform/module-ballerina-grpc/pull/192/commits/6c60129213e867a866d25b3c4afd3e48922812bb

File: grpc-compiler-plugin-tests/src/test/resources/test-src/package_10/grpc_service.bal [hunk 2]
```diff
@@ -0,0 +1,68 @@
+// This is the server implementation of the unary blocking/unblocking scenario.
```
Comment (daneshk) [inline grpc-compiler-plugin-tests/src/test/resources/test-src/package_10/grpc_service.bal:1]: please add license headers

File: grpc-compiler-plugin-tests/src/test/resources/test-src/package_11/grpc_service.bal [hunk 3]
```diff
@@ -0,0 +1,68 @@
+// This is the server implementation of the unary blocking/unblocking scenario.
```
Comment (daneshk) [inline grpc-compiler-plugin-tests/src/test/resources/test-src/package_11/grpc_service.bal:1]: add license header

File: grpc-compiler-plugin-tests/src/test/resources/test-src/package_10/grpc_service.bal (hunk 2 above)
Comment (BuddhiWathsala) [reply grpc-compiler-plugin-tests/src/test/resources/test-src/package_10/grpc_service.bal]: Fixed in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/192/commits/a3ea94e1cbe73cb9d935070ad3b3bd9a137a041d

File: grpc-compiler-plugin-tests/src/test/resources/test-src/package_11/grpc_service.bal (hunk 3 above)
Comment (BuddhiWathsala) [reply grpc-compiler-plugin-tests/src/test/resources/test-src/package_11/grpc_service.bal]: Fixed in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/192/commits/a3ea94e1cbe73cb9d935070ad3b3bd9a137a041d

**Review by daneshk — APPROVED** [review]
LGTM

**Pull request merged**